pkg/acceptance/helpers/information_schema_client.go

```diff
@@ -25,20 +27,37 @@ func (c *InformationSchemaClient) client() *sdk.Client {
 	return c.context.client
 }
 
-func (c *InformationSchemaClient) GetQueryTextByQueryId(t *testing.T, queryId string) string {
+type QueryHistory struct {
+	QueryId   string
+	QueryText string
+	QueryTag  string
+}
+
+func (c *InformationSchemaClient) GetQueryHistory(t *testing.T, limit int) []QueryHistory {
 	t.Helper()
-	result, err := c.client().QueryUnsafe(context.Background(), fmt.Sprintf("SELECT QUERY_TEXT FROM TABLE(INFORMATION_SCHEMA.QUERY_HISTORY(RESULT_LIMIT => 20)) WHERE QUERY_ID = '%s'", queryId))
+	result, err := c.client().QueryUnsafe(context.Background(), fmt.Sprintf("SELECT * FROM TABLE(INFORMATION_SCHEMA.QUERY_HISTORY(RESULT_LIMIT => %d))", limit))
 	require.NoError(t, err)
-	require.Len(t, result, 1)
-	require.NotNil(t, result[0]["QUERY_TEXT"])
-	return (*result[0]["QUERY_TEXT"]).(string)
+	return collections.Map(result, func(m map[string]*any) QueryHistory {
+		require.NotNil(t, m["QUERY_ID"])
+		require.NotNil(t, m["QUERY_TEXT"])
+		require.NotNil(t, m["QUERY_TAG"])
+		return QueryHistory{
+			QueryId:   (*m["QUERY_ID"]).(string),
+			QueryText: (*m["QUERY_TEXT"]).(string),
+			QueryTag:  (*m["QUERY_TAG"]).(string),
+		}
+	})
 }
 
-func (c *InformationSchemaClient) GetQueryTagByQueryId(t *testing.T, queryId string) string {
+func (c *InformationSchemaClient) GetQueryHistoryByQueryId(t *testing.T, limit int, queryId string) QueryHistory {
```

Comment:
I think we can agree on a default limit here and avoid specifying it in each call.

Reply:
I would leave it as a value to specify as I changed this a lot during testing and we should aim for as little number as possible